core/integration/storagetest/mutation_logs_reader.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetest
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/keytransparency/core/sequencer"
+
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
+)
+
+type LogsReader interface {
+	sequencer.LogsReader
+	// Send submits the whole group of mutations atomically to a random log.
+	// TODO(gbelvin): Create a batch level object to make it clear that this a batch of updates.
+	// Returns the logID and timestamp that the mutation batch got written at.
+	Send(ctx context.Context, directoryID string, logID int64, mutation ...*pb.EntryUpdate) (time.Time, error)
+}
+
+// mutationLogsReaderFactory returns a new database object, and a function for cleaning it up.
+type mutationLogsReaderFactory func(ctx context.Context, t *testing.T, dirID string, logIDs ...int64) (LogsReader, func(context.Context))
+
+// RunMutationLogsReaderTests runs all the tests against the provided storage implementation.
+func RunMutationLogsReaderTests(t *testing.T, factory mutationLogsReaderFactory) {
+	ctx := context.Background()
+	b := &mutationLogsReaderTests{}
+	for name, f := range map[string]func(ctx context.Context, t *testing.T, f mutationLogsReaderFactory){
+		// TODO(gbelvin): Discover test methods via reflection.
+		"TestHighWatermark": b.TestHighWatermark,
+	} {
+		t.Run(name, func(t *testing.T) { f(ctx, t, factory) })
+	}
+}
+
+type mutationLogsReaderTests struct{}
+
+// TestReadLogExact ensures that reads respect the low inclusive, high exclusive API.
```

Comment:
Please sync the comment and the function name (and `directoryID` if the name changes).

Reply:
I'm not sure why the linter doesn't pick this up...